tasks/node/purge.js

```diff
@@ -4,67 +4,46 @@ require('dotenv').config();
 	process.exit(1);
 }
 
-const flatten = require('lodash').flattenDeep;
-const qaHostNames = [
-	'http://qa.polyfill.io'
-];
-const productionHostNames = [
-	'https://polyfill.io',
-	'https://polyfill.webservices.ft.com',
-	'https://cdn.polyfill.io'
-];
-const hostnames = PRODUCTION ? productionHostNames : qaHostNames;
-const paths = [
-	'/v2/',
-	'/v2/assets/css/style.css',
-	'/v2/assets/images/fastly-logo.png',
-	'/v2/assets/images/logo.svg',
-	'/v2/assets/js/ui.js',
-	'/v2/docs/',
-	'/v2/docs/api',
-	'/v2/docs/contributing',
-	'/v2/docs/contributing/authoring-polyfills',
-	'/v2/docs/contributing/common-scenarios',
-	'/v2/docs/contributing/docs/assets/compat.json',
-	'/v2/docs/contributing/testing',
-	'/v2/docs/examples',
-	'/v2/docs/features/',
-	'/v2/docs/usage'
-];
-const endpoints = flatten(paths.map(path => hostnames.map(host => host + path)));
+if (PRODUCTION && !process.env.FASTLY_SERVICE_ID) {
+	console.error('In order to purge assets from the production Fastly service, you need to have set the environment variable "FASTLY_SERVICE_ID". This can be done by creating a file named ".env" in the root of this repository with the contents "FASTLY_SERVICE_ID=XXXXXX", where XXXXXX is your Fastly service ID.');
+	process.exit(1);
+} else {
+	FASTLY_SERVICE_ID = process.env.FASTLY_SERVICE_ID;
+}
 
-if (!DRY_RUN) {
-	console.log();
-	const fetch = require('node-fetch');
-	Promise.all(endpoints.map(endpoint => {
-		return fetch(endpoint, {
-			method: 'PURGE',
-			headers: {"Fastly-Key": process.env.FASTLY_API_KEY},
-			mode: 'no-cors',
-			cache: 'default'
-		})
-		.then(function(res) {
-			if (res.ok) {
-				console.log(`Purged ${endpoint}`);
-			} else if (res.status === 401){
-				throw Error('It seems you may not be authorised to purge the resources. Have you set the environment variable "FASTLY_API_KEY"?');
-			}
-		});
-	}))
-	.then(() => console.log('\nPurged all endpoints successfully.'))
-	.catch((e) => console.error(`Failed to purge endpoints. ${e}`));
+if (!PRODUCTION && !process.env.FASTLY_SERVICE_ID_QA) {
+	console.error('In order to purge assets from the QA Fastly service, you need to have set the environment variable "FASTLY_SERVICE_ID_QA". This can be done by creating a file named ".env" in the root of this repository with the contents "FASTLY_SERVICE_ID_QA=XXXXXX", where XXXXXX is your Fastly service ID.');
+	process.exit(1);
 } else {
-	console.log('\nThis is a dry run. No assets were purged from the cache.\n');
-	console.log('If this were not a dry run, these are the assets which would have been purged from the cache:\n');
-	endpoints.forEach(endpoint => console.log(endpoint));
+	FASTLY_SERVICE_ID = process.env.FASTLY_SERVICE_ID_QA;
 }
+
+const FastlyPurge = require('fastly-purge');
+
+const fastly = new FastlyPurge(FASTLY_API_KEY, {
+	softPurge: true
+});
+
+const key = 'polyfill-service';
```

Comment:
Maybe hoist this def to the top and name it as a constant, eg `CDN_SURROGATE_KEY`.  Also since this needs to be in sync between this task and the lib code, should it be an env var?